packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -455,6 +472,13 @@ abstract class BucketBase extends Resource implements IBucket {
     return `${this.bucketArn}/${keyPattern}`;
   }
 
+  /**
+   * Adds write permissions to the LogDelivery group via the Bucket ACL.
+   */
+  public allowLogDelivery() {
```

Comment:
Let's add a check that verifies that `accessControl` was not set to another value and fail in that case. Otherwise, this can introduce issues that are very hard to debug ("who changed my ACL?")

Reply:
I'm unsure on the Error message, let me know what you think about it